src/NCronJob/Registry/JobState.cs

```diff
@@ -0,0 +1,28 @@
+namespace NCronJob;
+
+internal class JobState
+{
+    public JobStateType Type { get; private set; }
+    public DateTime Timestamp { get; private set; }
+    public string Message { get; private set; }
```

Comment:
Maybe I am blind - where is the `Message` used?

Reply:
The message is currently only set when faulting so it uses the exception's message. My initial thought was that most of the logs and telemetry would reside within the `OnStateChanged` event handler so having the exception message could be valuable.
https://github.com/NCronJob-Dev/NCronJob/blob/9e45fc8fe348f818c2def27e50ac78d040c44ec6/src/NCronJob/Registry/JobRun.cs#L48-L50